### ISSUE TITLE
[TECHDEBT] [P2P] Complete P2P Module

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#275]
## Objective

Tend to this TODO:

```go
// TODO(team): This is a a temporary parallel to the real `p2p` module.
// It should be removed once the real `p2p` module is ready but is meant
// to be a "real" replacement for now.
```

We need to identify the definition of "ready" within the current milestone and update below accordingly

## Origin Document

[ Why? Issue justification and/or link to another document]

## Goals

- _\_*REPLACE_ME*: List of things that are high level ideas or goals driving the task_
- [ ] ...

## Deliverable

- _\_*REPLACE_ME*: List of things that are concrete deliverables_
- [ ] ...

## Non-goals / Non-deliverables

- _\_*REPLACE_ME*: List of things that are out of scope_
- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
This was an outdated comment and I believe I removed it in one of the inflight PRs.

We had a different p2p module in the past.

We're going to go with this one, but I think more importantly is creating a ticket to:
1. Research https://libp2p.io/
2. Summarize & present learnings to the team (plus what we should & shouldn't use from the library)
3. If we decide to move forward with it, a ticket to integrate it.

Could you create a discussion on GH to "Investigate whether libp2p should be used as the foundational network layer" and look into that? I can mark this one as "in progress" and we can close it out once those issues & discussions are created and the TODO is removed.

**Comment by jessicadaugherty** [comment]
Thank you @deblasis! Let me know if this should turn into an issue (research spike) at any point so the work stays visible on the board.

**Comment by deblasis** [comment]
@jessicadaugherty:
> Thank you @deblasis! Let me know if this should turn into an issue (research spike) at any point so the work stays visible on the board.

Yeah I think it should be tracked on the board and also, I would sadistically ask for a deadline so that I can plan all my other activities (life included :)) accordingly.

"work expands so as to fill the time available for its completion."

https://en.wikipedia.org/wiki/Parkinson%27s_law

**Comment by Olshansk** [comment]
@jessicadaugherty Can you link to the libp2p research ticket here once it's ready, and we'll document the results of that work in #287.

Once the ticket is created, we can close this one out.

**Comment by Olshansk** [comment]
Agreed. Closing per the comment above.

**Comment by Olshansk** [comment]
Note that this also involves some work related to the testing framework, redundancy layer, etc, but all of that work is captured in more well scoped tickets.
